src/ServiceControl/Operations/ImportFailedAudits.cs

```diff
@@ -0,0 +1,86 @@
+namespace ServiceControl.Operations
+{
+    using System;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using NServiceBus.Logging;
+    using Raven.Client;
+    using Raven.Client.Indexes;
+
+    public class ImportFailedAudits
+    {
+        public ImportFailedAudits(IDocumentStore store, AuditImporter auditImporter)
+        {
+            this.store = store;
+            this.auditImporter = auditImporter;
+        }
+
+        public Task Run(CancellationTokenSource tokenSource)
+        {
+            source = tokenSource;
+            return Task.Factory.StartNew(() => Run<FailedAuditImport, FailedAuditImportIndex>(source.Token));
+        }
+
+        void Run<T, I>(CancellationToken token) where I : AbstractIndexCreationTask, new()
+        {
+            var succeeded = 0;
+            var failed = 0;
+            using (var session = store.OpenSession())
+            {
+                var query = session.Query<T, I>();
+                using (var ie = session.Advanced.Stream(query))
+                {
+                    while (!token.IsCancellationRequested && ie.MoveNext())
+                    {
+                        FailedTransportMessage dto = ((dynamic)ie.Current.Document).Message;
+                        try
+                        {
+                            var transportMessage = new TransportMessage(dto.Id, dto.Headers)
+                            {
+                                Body = dto.Body
+                            };
+
+                            var entity = auditImporter.ConvertToSaveMessage(transportMessage);
+                            using (var storeSession = store.OpenSession())
+                            {
+                                storeSession.Store(entity);
+                                storeSession.SaveChanges();
+                            }
+                            store.DatabaseCommands.Delete(ie.Current.Key, null);
```

Comment:
The `ProcessedMessage` could be persisted to raven and before the key is deleted, we could fail. Are we ok with duplicate `ProcessedMessage`s?